### ISSUE TITLE
Same protein pairs but have different labels

### ISSUE THREAD

**Issue by YanjunLi-CS** [samsledje/D-SCRIPT#30]
Hi Sam,

Thanks for your awesome work!

When I checked the tsv file in the data/pairs folder, I found some protein pairs have the same protein IDs but shall different ground truth labels (0 and 1), which really confused me. Here I attach the samples found in the human_train.tsv.  

0 | 1 | 2
-- | -- | --
9606.ENSP00000296792 | 9606.ENSP00000254803 | 0
9606.ENSP00000284274 | 9606.ENSP00000352551 | 1
9606.ENSP00000321449 | 9606.ENSP00000369741 | 1
9606.ENSP00000340879 | 9606.ENSP00000369411 | 0
9606.ENSP00000282074 | 9606.ENSP00000389810 | 1
9606.ENSP00000389810 | 9606.ENSP00000282074 | 0
9606.ENSP00000360709 | 9606.ENSP00000303754 | 0
9606.ENSP00000254803 | 9606.ENSP00000296792 | 1
9606.ENSP00000301452 | 9606.ENSP00000313681 | 1
9606.ENSP00000367824 | 9606.ENSP00000467145 | 0
9606.ENSP00000335027 | 9606.ENSP00000348632 | 0
9606.ENSP00000368345 | 9606.ENSP00000482258 | 0
9606.ENSP00000378338 | 9606.ENSP00000422241 | 0
9606.ENSP00000300026 | 9606.ENSP00000316881 | 1
9606.ENSP00000217402 | 9606.ENSP00000324810 | 1
9606.ENSP00000230413 | 9606.ENSP00000320567 | 1
9606.ENSP00000468611 | 9606.ENSP00000378650 | 1
9606.ENSP00000384962 | 9606.ENSP00000365243 | 1
9606.ENSP00000342889 | 9606.ENSP00000460439 | 0
9606.ENSP00000388669 | 9606.ENSP00000367444 | 1
9606.ENSP00000296255 | 9606.ENSP00000347969 | 1
9606.ENSP00000354813 | 9606.ENSP00000443273 | 0
9606.ENSP00000443273 | 9606.ENSP00000354813 | 1
9606.ENSP00000422241 | 9606.ENSP00000378338 | 1
9606.ENSP00000369741 | 9606.ENSP00000378294 | 1
9606.ENSP00000347969 | 9606.ENSP00000296255 | 0
9606.ENSP00000358487 | 9606.ENSP00000408901 | 0
9606.ENSP00000316244 | 9606.ENSP00000248150 | 0
9606.ENSP00000303754 | 9606.ENSP00000360709 | 1
9606.ENSP00000316881 | 9606.ENSP00000300026 | 0
9606.ENSP00000369411 | 9606.ENSP00000340879 | 1
9606.ENSP00000229471 | 9606.ENSP00000288699 | 0
9606.ENSP00000320184 | 9606.ENSP00000223324 | 0
9606.ENSP00000305480 | 9606.ENSP00000055077 | 0
9606.ENSP00000315017 | 9606.ENSP00000308275 | 0
9606.ENSP00000367444 | 9606.ENSP00000388669 | 0
9606.ENSP00000223324 | 9606.ENSP00000320184 | 1
9606.ENSP00000443273 | 9606.ENSP00000346196 | 1
9606.ENSP00000460439 | 9606.ENSP00000342889 | 1
9606.ENSP00000324810 | 9606.ENSP00000217402 | 0
9606.ENSP00000380336 | 9606.ENSP00000263694 | 0
9606.ENSP00000453361 | 9606.ENSP00000385018 | 1
9606.ENSP00000352551 | 9606.ENSP00000284274 | 0
9606.ENSP00000230050 | 9606.ENSP00000363018 | 1
9606.ENSP00000343867 | 9606.ENSP00000468569 | 0
9606.ENSP00000250003 | 9606.ENSP00000242728 | 0
9606.ENSP00000385018 | 9606.ENSP00000453361 | 0
9606.ENSP00000378650 | 9606.ENSP00000468611 | 0
9606.ENSP00000408901 | 9606.ENSP00000358487 | 1
9606.ENSP00000348632 | 9606.ENSP00000335027 | 1
9606.ENSP00000339527 | 9606.ENSP00000366711 | 1
9606.ENSP00000248150 | 9606.ENSP00000316244 | 1
9606.ENSP00000482258 | 9606.ENSP00000368345 | 1
9606.ENSP00000320567 | 9606.ENSP00000230413 | 0
9606.ENSP00000242728 | 9606.ENSP00000250003 | 1
9606.ENSP00000288699 | 9606.ENSP00000229471 | 1
9606.ENSP00000324173 | 9606.ENSP00000288602 | 0
9606.ENSP00000386717 | 9606.ENSP00000461082 | 0
9606.ENSP00000308022 | 9606.ENSP00000362930 | 0
9606.ENSP00000288602 | 9606.ENSP00000324173 | 1
9606.ENSP00000365243 | 9606.ENSP00000384962 | 0
9606.ENSP00000313681 | 9606.ENSP00000301452 | 0
9606.ENSP00000278572 | 9606.ENSP00000378081 | 1
9606.ENSP00000369162 | 9606.ENSP00000375633 | 0
9606.ENSP00000263694 | 9606.ENSP00000380336 | 1
9606.ENSP00000376500 | 9606.ENSP00000385009 | 0
9606.ENSP00000308275 | 9606.ENSP00000315017 | 1
9606.ENSP00000366711 | 9606.ENSP00000339527 | 0
9606.ENSP00000448250 | 9606.ENSP00000285667 | 1
9606.ENSP00000482114 | 9606.ENSP00000330945 | 1
9606.ENSP00000378294 | 9606.ENSP00000369741 | 0
9606.ENSP00000429717 | 9606.ENSP00000294179 | 1
9606.ENSP00000270586 | 9606.ENSP00000264639 | 0
9606.ENSP00000428581 | 9606.ENSP00000320567 | 1
9606.ENSP00000320567 | 9606.ENSP00000428581 | 0
9606.ENSP00000461082 | 9606.ENSP00000386717 | 1
9606.ENSP00000467145 | 9606.ENSP00000367824 | 1
9606.ENSP00000385009 | 9606.ENSP00000376500 | 1
9606.ENSP00000055077 | 9606.ENSP00000305480 | 1
9606.ENSP00000330945 | 9606.ENSP00000482114 | 0
9606.ENSP00000264639 | 9606.ENSP00000270586 | 1
9606.ENSP00000362930 | 9606.ENSP00000308022 | 1
9606.ENSP00000346196 | 9606.ENSP00000443273 | 0
9606.ENSP00000363018 | 9606.ENSP00000230050 | 0
9606.ENSP00000378081 | 9606.ENSP00000278572 | 0
9606.ENSP00000375633 | 9606.ENSP00000369162 | 1
9606.ENSP00000468569 | 9606.ENSP00000343867 | 1
9606.ENSP00000363746 | 9606.ENSP00000388126 | 0
9606.ENSP00000388126 | 9606.ENSP00000363746 | 1
9606.ENSP00000285667 | 9606.ENSP00000448250 | 0
9606.ENSP00000294179 | 9606.ENSP00000429717 | 0
9606.ENSP00000369741 | 9606.ENSP00000321449 | 0

Also, similar cases happen in the yeast_test.tsv:

0 | 1 | 2
-- | -- | --
4932.YBR159W | 4932.YPL220W | 1
4932.YMR230W | 4932.YHR021C | 0
4932.YJL069C | 4932.YPR137W | 1
4932.YJL033W | 4932.YLL011W | 1
4932.YHR021C | 4932.YMR230W | 1
4932.YLL011W | 4932.YJL033W | 0
4932.YPL220W | 4932.YBR159W | 0
4932.YPR137W | 4932.YJL069C | 0

Could you please check these protein pairs and verify their ground truth? Thank you very much!

**Comment by rs239** [comment]
Hi YanjunLi-CS,
 Thanks for finding this-- we'll investigate it. I'm pretty sure these
should be all positive. Here's how we generated the labels:
  a) positive labels only come from STRING, with experimental evidence for
PPI
  b) negative labels are generated by randomly choosing pairs of proteins.
We have code that checks to see if such a pair doesn't already have a
positive label before outputting it. Seems like there's a corner case in
that check we missed.

To summarize: as long as a pair has a "1" label with it, you should treat
it as ground truth positive and ignore the "0" labels. Thanks for reporting
the bug!

best
-r

**Comment by YanjunLi-CS** [comment]
Hi rs239,

Thanks for your prompt reply and confirmation! That is very helpful! 

BTW, I notice that the human data files for training and testing seem to be updated two times. If we would like to train and test on the same data set used by D-script, whether we should use this latest version (Feb 7, 2022) and remove the above conflict negative samples. Thank you!

**Comment by samsledje** [comment]
Hi YanjunLi-CS,

There was an issue in the files that were originally updated -- I would recommend using the latest version of the files as current in the repository.
